src/main/java/org/purpurmc/purpurextras/modules/LeashSnapSoundModule.java

```diff
@@ -0,0 +1,55 @@
+package org.purpurmc.purpurextras.modules;
+
+import net.kyori.adventure.key.Key;
+import net.kyori.adventure.sound.Sound;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityUnleashEvent;
+import org.purpurmc.purpurextras.PurpurExtras;
+
+import java.util.Locale;
+
+/**
+ * Adds a sound for when the leash snaps
+ */
+public class LeashSnapSoundModule implements PurpurExtrasModule, Listener {
+
+    private Key sound;
+    private double volume;
+    private double pitch;
+
+    protected LeashSnapSoundModule() {
+        String soundId = PurpurExtras.getPurpurConfig().getString("settings.leash-snap.sound", "block.bamboo.break");
+
+        try {
+            sound = org.bukkit.Sound.valueOf(soundId.replace('.', '_').toUpperCase(Locale.ROOT)).key();
```

Comment:
Sound enum does not cover sounds added via resource packs and thus will fail to resolve on the sounds not present in the enum. Create a key normally without an enum and use the playSound with string arg instead of the Sound enum